spring-pulsar/src/test/java/org/springframework/pulsar/listener/PulsarListenerTests.java

```diff
@@ -955,4 +957,53 @@ void pulsarMessagesBatchListenerWithHeaders(Messages<String> data,
 
 	}
 
+	@Nested
+	@ContextConfiguration(classes = ConsumerPauseTest.ConsumerPauseConfig.class)
+	class ConsumerPauseTest {
+
+		private static final CountDownLatch latch = new CountDownLatch(10);
+
+		@Autowired
+		PulsarListenerEndpointRegistry pulsarListenerEndpointRegistry;
+
+		@Test
+		void containerPauseAndResumeSuccessfully() throws Exception {
+			for (int i = 0; i < 3; i++) {
+				pulsarTemplate.send("consumer-pause-topic", "hello-" + i);
+			}
+			// wait until all 3 messages are received by the listener
+			Awaitility.await().timeout(Duration.ofSeconds(10)).until(() -> latch.getCount() == 7);
+			PulsarMessageListenerContainer container = pulsarListenerEndpointRegistry
+					.getListenerContainer("consumerPauseListener");
+			if (container != null) {
```

Comment:
We should probably assertThat it is not null prior to this line.

Reply:
ok